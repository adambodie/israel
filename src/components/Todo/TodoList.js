import React from 'react';
import { Row, Column } from 'react-foundation';
import ReactMarkdown from 'react-markdown';
import TodoListForm from './TodoListForm.js';
import TodoListItemList from './TodoListItemList.js';
import { intro } from '../../data/main';
import { Provider } from '../Context/todo';
import ReactTooltip from 'react-tooltip'


const TodoList = () => {	
	return(
		<Provider>
			<Row className="display grid-x todoList">
				<Column large={6} offsetOnLarge={3}>
					<h1>Things to Bring to Israel</h1>
					<ReactMarkdown source={intro} className="todoListIntro"/>
					<img src={'https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/items.jpg'} alt='bunch of items' data-tip="A Bunch of Items (Not all of these items are needed for this trip)"/>
					<ReactTooltip />
					<ul>
						<TodoListItemList />
					</ul>
					<TodoListForm />
				</Column>
			</Row>
		</Provider>	
	)
}

export default TodoList;


