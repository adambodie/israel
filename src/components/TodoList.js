import React, { Component} from 'react';
import TodoListForm from './TodoListForm.js';
import TodoListItem from './TodoListItem.js';

export default class TodoList extends Component {
	constructor(){
		super();
		this.state = {
			items: [{name: 'Shirts', id: 1, className: 'item', count: 0},{name:'Pants', id: 2, className: 'item', count: 0}, {name:'Passport', id: 3, className: 'item', count: 0}, {name: 'Shoes', id: 4, className: 'item', count: 0}, {name:'Camera', id: 5, className: 'item', count: 0}],
			value: '',
		}
		this.onHandleChange = this.onHandleChange.bind(this);
		this.onHandleSubmit = this.onHandleSubmit.bind(this);		
		this.onHandleRemove = this.onHandleRemove.bind(this);		
		this.onHandleComplete = this.onHandleComplete.bind(this);		
	}
	onHandleChange(value) {
		this.setState({value});
	}	
	onHandleComplete(index) {
		const newItems = [...this.state.items];
		let count = newItems[index - 1].count;
		count++;
		if (count % 2 === 1) {
			newItems.forEach((item) => {
				if (item.id === index) {
					item.className = "item-completed";
					item.count++;
				}				
				
			})
		} else {
			newItems.forEach((item) => {
				if (item.id === index) {
					item.className = "item";
					item.count++
				}
			})
		}
		this.setState({newItems});		
	}

	onHandleSubmit() {
		const newItems = [...this.state.items];
		if (this.state.value !== '') {
			newItems.push({name:this.state.value, id: this.state.items.length + 1, className: 'item', count: 0});
			this.setState(prevState =>({ 
				items: newItems, 
				value: '' 
			}));
		} else {
			alert("Please enter a value");
		};
	}
	onHandleRemove(index) {
		const newItems = [...this.state.items];
		newItems.splice(index - 1 , 1);
		newItems.forEach((item, index) => item.id = index + 1);
		this.setState(prevState =>({ items: newItems }));
	}		
	render() {		
		let listOfOtherItems = this.state.items.map((x, index) => {
			return(
				<TodoListItem id={x.id} 
							  key={x.id} 
							  className={x.className} 
							  name={x.name} 
							  onHandleRemove={this.onHandleRemove} 
							  onHandleComplete={this.onHandleComplete} 
							/>
					)
				}
			);
		return(
			<div>
				<h1>Things to Bring to Israel</h1>
					<ul className="todoListEdit">
						{listOfOtherItems}
					</ul>
					<TodoListForm onHandleChange={this.onHandleChange} onHandleSubmit={this.onHandleSubmit} value={this.state.value} />
			</div>
		)
  }
}


