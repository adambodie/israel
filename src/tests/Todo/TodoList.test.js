import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import TodoList from '../../components/Todo/TodoList';
import { MemoryRouter } from 'react-router';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<MemoryRouter><TodoList /></MemoryRouter>, div);
	ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () => {
	const tree = renderer.create(<MemoryRouter><TodoList /></MemoryRouter>).toJSON();
	expect(tree).toMatchSnapshot();
});

describe('<TodoList />', () => {
    const wrapper = mount(<MemoryRouter><TodoList /></MemoryRouter>);
    
    /*it('renders eight scrollspy items', () => {	
        expect(wrapper.find('.c-side-nav__item').length).toEqual(8);
     });

     it('renders nine sections', () => {	
        expect(wrapper.find('section').length).toEqual(9);
     });*/     
});