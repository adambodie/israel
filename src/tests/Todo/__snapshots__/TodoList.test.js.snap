// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row display grid-x todoList"
>
  <div
    className="columns large-6 large-offset-3"
  >
    <h1>
      Things to Bring to Israel
    </h1>
    <div
      className="todoListIntro"
    >
      <p>
        Before going to Israel, one must pack items to take with them.  Some things are automatic, like clothing.  Other things are required to enter, like your Passport.  Everything else is optional depending on your taste.  With that, feel free to add, remove and check off any items you'd like to bring!
      </p>
    </div>
    <img
      alt="bunch of items"
      data-tip="A Bunch of Items (Not all of these items are needed for this trip)"
      src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/items.jpg"
    />
    <div
      className="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    />
    <ul>
      <li
        className="item"
      >
        <a
          className="button delete alert"
          onClick={[Function]}
        >
          X
        </a>
        <p>
          Shirts
        </p>
        <a
          className="button complete primary"
          onClick={[Function]}
        >
          ✓
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="button delete alert"
          onClick={[Function]}
        >
          X
        </a>
        <p>
          Pants
        </p>
        <a
          className="button complete primary"
          onClick={[Function]}
        >
          ✓
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="button delete alert"
          onClick={[Function]}
        >
          X
        </a>
        <p>
          Passport
        </p>
        <a
          className="button complete primary"
          onClick={[Function]}
        >
          ✓
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="button delete alert"
          onClick={[Function]}
        >
          X
        </a>
        <p>
          Shoes
        </p>
        <a
          className="button complete primary"
          onClick={[Function]}
        >
          ✓
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="button delete alert"
          onClick={[Function]}
        >
          X
        </a>
        <p>
          Camera
        </p>
        <a
          className="button complete primary"
          onClick={[Function]}
        >
          ✓
        </a>
      </li>
    </ul>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        className="button"
        type="submit"
        value="Add Item"
      />
    </form>
    <div
      className="prev-next"
    >
      <a
        className="custom-button"
        href="/index"
        onClick={[Function]}
      >
        Prev
      </a>
      <a
        className="custom-button"
        href="/el-al"
        onClick={[Function]}
      >
        Next
      </a>
    </div>
  </div>
</div>
`;
